org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/PosDetectorScanTest.java

```diff
@@ -0,0 +1,225 @@
+import static org.eclipse.scanning.test.scan.nexus.NexusAssert.assertIndices;
+import static org.eclipse.scanning.test.scan.nexus.NexusAssert.assertScanNotFinished;
+import static org.eclipse.scanning.test.scan.nexus.NexusAssert.assertSignal;
+import static org.eclipse.scanning.test.scan.nexus.NexusAssert.assertSolsticeScanGroup;
+import static org.eclipse.scanning.test.scan.nexus.NexusAssert.assertTarget;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.eclipse.dawnsci.analysis.api.tree.DataNode;
+import org.eclipse.dawnsci.analysis.api.tree.TreeFile;
+import org.eclipse.dawnsci.nexus.NXcollection;
+import org.eclipse.dawnsci.nexus.NXdata;
+import org.eclipse.dawnsci.nexus.NXdetector;
+import org.eclipse.dawnsci.nexus.NXentry;
+import org.eclipse.dawnsci.nexus.NXinstrument;
+import org.eclipse.dawnsci.nexus.NXpositioner;
+import org.eclipse.dawnsci.nexus.NXroot;
+import org.eclipse.dawnsci.nexus.NexusFile;
+import org.eclipse.dawnsci.nexus.NexusUtils;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.january.dataset.PositionIterator;
+import org.eclipse.scanning.api.device.AbstractRunnableDevice;
+import org.eclipse.scanning.api.device.IRunnableDevice;
+import org.eclipse.scanning.api.device.IWritableDetector;
+import org.eclipse.scanning.api.device.models.IDetectorModel;
+import org.eclipse.scanning.api.event.scan.DeviceState;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.PositionEvent;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.event.IPositionListenable;
+import org.eclipse.scanning.api.scan.event.IPositionListener;
+import org.eclipse.scanning.api.scan.models.ScanModel;
+import org.eclipse.scanning.example.detector.PosDetectorModel;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PosDetectorScanTest extends NexusTest {
+	
+	private IWritableDetector<PosDetectorModel> detector;
+	
+	@Before
+	public void before() throws Exception {
+		PosDetectorModel model = new PosDetectorModel(3);
+		model.setTimeout(100); // TODO remove
+		detector = (IWritableDetector<PosDetectorModel>) dservice.createRunnableDevice(model);
+		assertNotNull(detector);
+	}
+	
+	@After
+	public void after() throws Exception {
+		File parentDir = output.getParentFile();
+		String fileName = output.getName().substring(0, output.getName().indexOf('.'));
+		File outputDir = new File(parentDir, fileName);
+		for (File file : outputDir.listFiles()) {
+//			file.delete();
+		}
+//		outputDir.delete();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)